java/src/main/java/ai/onnxruntime/TensorInfo.java

```diff
@@ -120,6 +120,20 @@ public static OnnxTensorType mapFromJavaType(OnnxJavaType type) {
     this.onnxType = onnxType;
   }
 
+  /**
+   * Constructs a TensorInfo with the specified shape and native type int.
+   *
+   * <p>Called from JNI.
+   *
+   * @param shape The tensor shape.
+   * @param typeInt The native type int.
+   */
+  TensorInfo(long[] shape, int typeInt) {
+    this.shape = shape;
```

Comment:
>  this.shape = shape; [](http://example.com/codeflow?start=2&length=21)

Also metadata does not really include a literal shape. It may have symbolic values. Real numeric shape is inferred at runtime and as such may not be a part of the info.
[https://github.com/onnx/onnx/blob/main/onnx/onnx-ml.proto#L648](https://github.com/onnx/onnx/blob/main/onnx/onnx-ml.proto#L648)

Reply:
See this comment https://github.com/microsoft/onnxruntime/pull/12516#discussion_r960031262. I could split them into two separate classes for each use case, but that seems wasteful as the information is mostly duplicated, or I could try and push the size information into the ONNX value classes themselves. But both of those are a more substantial refactor as it'll touch all the Java side stuff too, and I'm not convinced of the value of that.

Reply:
This way it roughly mirrors the TypeInfo class and the information that contains.

Reply:
That's the issue. I can not mirror that. The proto shape may include symbolic entries (strings) and it may potentially have a dimension entry that is empty. Such shapes denote the rank, but may not have an information on the dimension. This way we can match all shapes of the same rank.

Reply:
Ok, but that's a different refactor to this one, as I'll need to allow support for named dimensions in this object. At the moment I think they are just returned as -1 in the input/output info, as that's what `GetDimensions` returns.